plumbing/protocol/packp/advrefs.go

```diff
@@ -209,3 +199,27 @@ func (a *AdvRefs) IsEmpty() bool {
 		len(a.Peeled) == 0 &&
 		len(a.Shallows) == 0
 }
+
+// MakeReferenceSlice returns a sorted slice with all the references, their
+// peeled values, and symrefs.
+func (a *AdvRefs) MakeReferenceSlice() ([]*plumbing.Reference, error) {
+	allRefs := make([]*plumbing.Reference, 0)
+	refs, err := a.AllReferences()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Pre-allocate capacity:
```suggestion
	refs, err := a.AllReferences()
	if err != nil {
		return nil, err
	}
	allRefs := make([]*plumbing.Reference, 0, len(refs))
```